src/c-writer.cc

```diff
@@ -3043,15 +3044,15 @@ void CWriter::BeginFunction(const Func& func) {
   Write(Newline());
 }
 
-void CWriter::FinishFunction() {
+void CWriter::FinishFunction(size_t stack_var_section) {
   for (size_t i = 0; i < func_sections_.size(); ++i) {
     auto& [condition, stream] = func_sections_.at(i);
     std::unique_ptr<OutputBuffer> buf = stream.ReleaseOutputBuffer();
     if (condition.empty() || func_includes_.count(condition)) {
       stream_->WriteData(buf->data.data(), buf->data.size());
     }
 
-    if (i == 0) {
+    if (i == stack_var_section) {
```

Comment:
Is the stack_var_section not always the first one in a given function?

Reply:
not with these changes. we need to write `volatile` or not based on whether the function uses `try` or not, and we do that by pushing more func sections where relevant. in particular, we save the `stack_var_section` *after* writing params and locals (both of which may contain `volatile`), so we can put the stack vars in the correct place. (also, stack vars don't need `volatile` as any reused stack slots are written over after `longjmp`.)